packages/javascript/bh-shared-ui/src/hooks/useFileIngest.tsx

```diff
@@ -0,0 +1,61 @@
+import { useMutation, useQuery, useQueryClient } from 'react-query';
+import { apiClient } from '../utils/api';
+
+export const listFileIngestJobs = (skip?: number, limit?: number, sortBy?: string) =>
+    apiClient.listFileIngestJobs(skip, limit, sortBy).then((res) => res.data);
+
+export const listFileTypesForIngest = () => apiClient.listFileTypesForIngest().then((res) => res.data);
+
+export const startFileIngestJob = () => apiClient.startFileIngest().then((res) => res.data);
+
+export const uploadFileToIngestJob = ({
+    jobId,
+    fileContents,
+    contentType = 'application/json',
+}: {
+    jobId: string;
+    fileContents: any;
+    contentType?: string;
+}) => {
+    return apiClient.uploadFileToIngestJob(jobId, fileContents, contentType).then((res) => res.data);
+};
+
+export const endFileIngestJob = ({ jobId }: { jobId: string }) =>
+    apiClient.endFileIngest(jobId).then((res) => res.data);
+
+export const fileUploadKeys = {
+    all: ['file-upload'] as const,
```

Comment:
Is the idea around having this as an array so that we can add to it later and itll automatically be spread to the functions below? If we dont think we will need to add more keys, then I think we could set this as a string and not have to spread the array below.

Reply:
I don't think this is necessary, I've removed it